conans/client/graph/graph_binaries.py

```diff
@@ -209,11 +206,20 @@ def _evaluate_node(self, node, build_mode, update, remotes):
                 # package_id was not locked, this means a base lockfile that is being completed
                 locked.complete_base_node(node.package_id, node.prev)
 
+        if (node.binary in (BINARY_BUILD, BINARY_MISSING) and node.conanfile.info.invalid and
```

Comment:
This is the other important change: The first PACKAGE_ID_INVALID is not set directly, only after the package has been processed and it was detected that it is necessary to build it or that it is missing (no fallback to compatible packages, no removal of info in package_id()). As this is delayed, it allows to work without all settings defined if they are not part of the package_id, and the binary already exists.